provider/datagen/src/crabbake.rs

```diff
@@ -95,21 +104,18 @@ impl DataExporter<CrabbakeMarker> for ConstExporter {
         payload: DataPayload<CrabbakeMarker>,
     ) -> Result<(), DataError> {
         let (payload, marker_type) = payload.tokenize(&self.dependencies);
+        let payload_string = payload.to_string();
         let mut map = self.data.lock().expect("poison");
         if !map.contains_key(&key) {
             map.insert(key, (marker_type.to_string(), LiteMap::new()));
         }
-        map.get_mut(&key)
-            .unwrap()
-            .1
-            .insert(options, payload.to_string());
+        map.get_mut(&key).unwrap().1.insert(options, payload_string);
         Ok(())
     }
 
     fn flush(&self, key: ResourceKey) -> Result<(), DataError> {
-        if let Some((marker, raw)) = self.data.lock().expect("poison").remove(&key) {
-            let marker = syn::parse_str::<syn::Path>(&marker).unwrap();
-
+        let tmp = self.data.lock().expect("poison").remove(&key);
+        if let Some((marker, raw)) = tmp {
```

Comment:
@Manishearth I *believe* I have to do this because otherwise the lock will be held until the end of `flush`, but NLL will fix this in the future?